client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/MlClientDocumentationIT.java

```diff
@@ -2802,6 +2786,298 @@ public void onFailure(Exception e) {
+                }
+            };
+            // end::get-data-frame-analytics-execute-listener
+
+            // Replace the empty listener by a blocking listener in test
+            CountDownLatch latch = new CountDownLatch(1);
+            listener = new LatchedActionListener<>(listener, latch);
+
+            // tag::get-data-frame-analytics-execute-async
+            client.machineLearning().getDataFrameAnalyticsAsync(request, RequestOptions.DEFAULT, listener); // <1>
+            // end::get-data-frame-analytics-execute-async
+
+            assertTrue(latch.await(30L, TimeUnit.SECONDS));
+        }
+    }
+
+    public void testGetDataFrameAnalyticsStats() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+        client.machineLearning().putDataFrameAnalytics(new PutDataFrameAnalyticsRequest(DF_ANALYTICS_CONFIG), RequestOptions.DEFAULT);
+        {
+            // tag::get-data-frame-analytics-stats-request
+            GetDataFrameAnalyticsStatsRequest request = new GetDataFrameAnalyticsStatsRequest("my-analytics-config"); // <1>
+            // end::get-data-frame-analytics-stats-request
+
+            // tag::get-data-frame-analytics-stats-execute
+            GetDataFrameAnalyticsStatsResponse response =
+                client.machineLearning().getDataFrameAnalyticsStats(request, RequestOptions.DEFAULT);
+            // end::get-data-frame-analytics-stats-execute
+
+            // tag::get-data-frame-analytics-stats-response
+            List<DataFrameAnalyticsStats> stats = response.getAnalyticsStats();
+            // end::get-data-frame-analytics-stats-response
+
+            assertThat(stats.size(), equalTo(1));
+        }
+        {
+            GetDataFrameAnalyticsStatsRequest request = new GetDataFrameAnalyticsStatsRequest("my-analytics-config");
+
+            // tag::get-data-frame-analytics-stats-execute-listener
+            ActionListener<GetDataFrameAnalyticsStatsResponse> listener = new ActionListener<>() {
+                @Override
+                public void onResponse(GetDataFrameAnalyticsStatsResponse response) {
+                    // <1>
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    // <2>
+                }
+            };
+            // end::get-data-frame-analytics-stats-execute-listener
+
+            // Replace the empty listener by a blocking listener in test
+            CountDownLatch latch = new CountDownLatch(1);
+            listener = new LatchedActionListener<>(listener, latch);
+
+            // tag::get-data-frame-analytics-stats-execute-async
+            client.machineLearning().getDataFrameAnalyticsStatsAsync(request, RequestOptions.DEFAULT, listener); // <1>
+            // end::get-data-frame-analytics-stats-execute-async
+
+            assertTrue(latch.await(30L, TimeUnit.SECONDS));
+        }
+    }
+
+    private static final DataFrameAnalyticsConfig DF_ANALYTICS_CONFIG =
```

Comment:
#nit this seems like a weird ordering, having a `private static final` field between two `public` instance methods.